### PR TITLE
dagster: support dagster 1.5.x

### PR DESCRIPTION
**Pull request by tsungchih**
### Problem

The dagster integration needs to get event records by calling `DagsterInstance.get_event_records()` method with a `EventRecordsFilter`. The current dagster integration has been broken since Dagster version [0.15.0](https://github.com/dagster-io/dagster/releases/tag/0.15.0) due to API change on `EventRecordsFilter` which requires a `DagsterEventType`.

Closes: #2043 

### Solution

In the factory method `openlineage_sensor`, an additional argument `concerned_event_types` is added with default value including events in `PIPELINE_EVENTS` and `STEP_EVENTS`. When getting event records, the dagster integration constructs `EventRecordsFilter` for each `DagsterEventType` embraced in the given `concerned_event_types` argument. The results will be consolidated and sorted against `timestamp` of event record. 

> **Note:** All schema changes require discussion. Please [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) for context.

- [ ] Your change modifies the [core](https://github.com/OpenLineage/OpenLineage/blob/main/spec/OpenLineage.json) OpenLineage model
- [ ] Your change modifies one or more OpenLineage [facets](https://github.com/OpenLineage/OpenLineage/tree/main/spec/facets)

If you're contributing a new integration, please specify the scope of the integration and how/where it has been tested (e.g., Apache Spark integration supports `S3` and `GCS` filesystem operations, tested with AWS EMR).

### Checklist

- [X] You've [signed-off](https://github.com/OpenLineage/OpenLineage/blob/main/why-the-dco.md) your work
- [X] Your pull request title follows our [guidelines](https://github.com/OpenLineage/OpenLineage/blob/main/CONTRIBUTING.md#creating-pull-requests)
- [X] Your changes are accompanied by tests (_if relevant_)
- [X] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [X] You've updated any relevant documentation (_if relevant_)
- [X] Your comment includes a one-liner for the changelog about the specific purpose of the change (_if necessary_)
- [ ] You've versioned the core OpenLineage model or facets according to [SchemaVer](https://docs.snowplowanalytics.com/docs/pipeline-components-and-applications/iglu/common-architecture/schemaver) (_if relevant_)
- [ ] You've added a [header](https://github.com/OpenLineage/OpenLineage/tree/main/.github/header_templates.md) to source files (_if relevant_)

----
SPDX-License-Identifier: Apache-2.0\
Copyright 2018-2023 contributors to the OpenLineage project

### CODE REVIEW

**Comment by JDarDagran** [comment]
@tsungchih  would you like to rebase to HEAD origin/main? Current CI process detects changes between latest main and dev branch so it runs pipelines for Spark, Airflow etc.

**Comment by tsungchih** [comment]
@JDarDagran Allow me to make myself clear. Do you mean to execute `git rebase HEAD origin/main`?

**Comment by JDarDagran** [comment]
```shell
# Make sure your local branch is up to date with the remote main branch
git fetch origin

# Rebase your local commits on top of the latest commits from the remote main branch
git rebase origin/main

# Push rebased commits
git push my-branch --force-with-lease
```

**Comment by tsungchih** [comment]
Many thanks for the guidance. I ran the following commands and there's nothing changed. Am I missing something??

```shell
❯ git fetch origin
❯ git rebase origin/main
Current branch dagster/support-dagster-1.5.x is up to date.
❯ git push --force-with-lease origin dagster/support-dagster-1.5.x
Everything up-to-date
```

File: integration/dagster/openlineage/dagster/sensor.py [hunk 1]
```diff
@@ -28,13 +28,15 @@ def openlineage_sensor(
         description: Optional[str] = "OpenLineage sensor tails Dagster event logs, "
                                      "converts Dagster events into OpenLineage events, "
                                      "and emits them to an OpenLineage backend.",
+        concerned_event_types: Optional[Union[DagsterEventType, Set[DagsterEventType]]] = PIPELINE_EVENTS | STEP_EVENTS, # noqa: E501
```
Comment (JDarDagran) [inline integration/dagster/openlineage/dagster/sensor.py:31]: Why this is optional?
In `get_event_log_records` it's just
`event_type: Union[DagsterEventType, Set[DagsterEventType]]`

**Review by JDarDagran — APPROVED** [review]
@tsungchih sorry for the delay in review. It's a well done job! Just one small comment.
Thank you!

File: integration/dagster/openlineage/dagster/sensor.py (hunk 1 above)
Comment (tsungchih) [reply integration/dagster/openlineage/dagster/sensor.py:31]: @JDarDagran I am sorry to make you confused. I forgot to remove that `Optional`. Originally `concerned_event_types` was designed as `Optional`, the default value (`PIPELINE_EVENTS | STEP_EVENTS`) was given when calling `get_event_log_records` if that argument is `None`. To get users to know the default value without digging into the source code, I decided to set up the default value in the factory function parameter but forgot to remove `Optional`.

Here's the code snippets of original implementation.

```python
def openlineage_sensor(
        name: Optional[str] = "openlineage_sensor",
        description: Optional[str] = "OpenLineage sensor tails Dagster event logs, "
                                     "converts Dagster events into OpenLineage events, "
                                     "and emits them to an OpenLineage backend.",
        concerned_event_types: Optional[Union[DagsterEventType, Set[DagsterEventType]]] = None,
        minimum_interval_seconds: Optional[int] = DEFAULT_SENSOR_DAEMON_INTERVAL,
        record_filter_limit: Optional[int] = 30,
        after_storage_id: Optional[int] = 0
) -> SensorDefinition:
    @sensor(    # type: ignore
        name=name,
        minimum_interval_seconds=minimum_interval_seconds,
        description=description,
    )
   def _openlineage_sensor(context: SensorEvaluationContext):
        ...
        ...
        event_log_records = get_event_log_records(
            context.instance, concerned_event_types or PIPELINE_EVENTS | STEP_EVENTS, last_storage_id, record_filter_limit
        )
        ...
        ...
```

Could I correct this in my local branch and push back to the remote branch??

Comment (JDarDagran) [reply integration/dagster/openlineage/dagster/sensor.py:31]: Yes, please 👍

**Comment by tsungchih** [comment]
@JDarDagran I saw the CI pipeline is failed. Did I mess up something? Wanna know where the problem is for my operation (pull -> merge -> commit -> push) so that I won't make mistake again.

**Comment by JDarDagran** [comment]
@tsungchih main was broken. branch needed rebase, I let myself do it.

Merging it, thank you so much for you cooperation and contribution! 🚀

**Comment by tsungchih** [comment]
@JDarDagran Many thanks! Do you mean I should perform rebase instead of merge when I was pulling changes from the remote branch??

**Comment by JDarDagran** [comment]
Rebase/merge is quite the same in this case for me. I've applied changes from origin/main so the CI doesn't fail.

**Pull request merged**